### PR TITLE
Deploy app as multi-container service to EC2

### PR DESCRIPTION
**Pull request by cywang117**
- DB + server containers on the same network via docker-compose
- Deployed to EC2 using development NODE_ENV (for now, just to make
sure it works. Production NODE_ENV requires some further tweaking to
work with Docker)

@WDHudson The changes here are mostly Docker config files & database config tweaking to work with Docker. You can find the deployed link [here](http://ec2-18-224-215-81.us-east-2.compute.amazonaws.com:3001/). It's HTTP instead of HTTPS right now though, so still some tweaking to do, but this is proof of concept. Thanks!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ENV NODE_ENV=development
+ENV PGDB_URI=postgresql://postgres:postgres@db:5432/steam_reviews
+
+COPY package*.json ./
+RUN npm install
+# RUN npm ci --only=production
+COPY . .
+EXPOSE 3001
+RUN npm run build
+CMD npm start
```
Comment (WDHudson) [inline Dockerfile:19]: Can this work not in array form?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - '5432:5432'
+    networks:
+      - reviews-network
+
+networks:
+  reviews-network:
+    driver: bridge
+
+volumes:
+  pgdata:
```
Comment (WDHudson) [inline docker-compose.yml:41]: Looks good!

File: wait-for.sh [hunk 3]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+# Use this script to test if a given TCP host/port are available
+
```
Comment (WDHudson) [inline wait-for.sh:3]: I like the clear pseudocode to describe what's going on.  It nice to have a question answered just as you're asking it to yourself....

File: Dockerfile (hunk 1 above)
Comment (cywang117) [reply Dockerfile:19]: I think it works both ways

**Pull request merged**